service/storage_service.cc

```diff
@@ -3058,10 +3059,10 @@ storage_service::describe_ring(const sstring& keyspace, bool include_only_local_
     std::vector<token_range_endpoints> ranges;
     //Token.TokenFactory tf = getPartitioner().getTokenFactory();
 
-    std::unordered_map<dht::token_range, inet_address_vector_replica_set> range_to_address_map =
-            include_only_local_dc
+    std::unordered_map<dht::token_range, inet_address_vector_replica_set> range_to_address_map = co_await
+            (include_only_local_dc
                     ? get_range_to_address_map_in_local_dc(keyspace)
-                    : get_range_to_address_map(keyspace);
+                    : get_range_to_address_map(keyspace));
     auto tmptr = get_token_metadata_ptr();
```

Comment:
Before this change, `range_to_address_map` was coherent with `tmptr`.

After, another call can interleave. I don't know if it's a problem or not, but better not to risk it.

All these topology things must be done transactionally: the top-level caller needs to open a transaction (acquire a tmptr), operate on the snapshot (do all calls using a tmptr) and then close the transaction (release the tmptr).

Long running operations like repair can listen to changes in topology and react to them (maybe by aborting repair) but the minimum is not to perform calls that can use different generations of tmptr.

Reply:
Fixed in v2